api/desecapi/tests/test_pdns_change_tracker.py

```diff
@@ -0,0 +1,461 @@
+        rr.save()
+        rr.delete()
+
+    def test_create_update_delete_empty_rr_set_2(self):
+        with self.assertPdnsEmptyRRSetUpdate(), PDNSChangeTracker():
+            RR.objects.create(rrset=self.empty_rr_set, content=self.CONTENT_VALUES[0])
+            rr = RR.objects.create(rrset=self.empty_rr_set, content=self.CONTENT_VALUES[1])
+            rr.content = self.ALT_CONTENT_VALUES[1]
+            rr.save()
+            RR.objects.create(rrset=self.empty_rr_set, content=self.CONTENT_VALUES[2])
+            rr.delete()
+
+    def test_create_update_delete_simple_rr_set(self):
+        with self.assertPdnsSimpleRRSetUpdate(), PDNSChangeTracker():
+            self.simple_rr_set.records.all()[0].delete()
+            RR.objects.create(rrset=self.simple_rr_set, content=self.CONTENT_VALUES[0])
+            rr = RR.objects.create(rrset=self.simple_rr_set, content=self.CONTENT_VALUES[1])
+            rr.content = self.ALT_CONTENT_VALUES[1]
+            rr.save()
+
+    def test_create_update_delete_full_rr_set(self):
+        with self.assertPdnsFullRRSetUpdate(), PDNSChangeTracker():
+            self.full_rr_set.records.all()[1].delete()
+            rr = self.full_rr_set.records.all()[1]
+            rr.content = self.ALT_CONTENT_VALUES[0]
+            rr.save()
+            RR.objects.create(rrset=self.full_rr_set, content=self.ALT_CONTENT_VALUES[1])
+
+
+class AAAARRTestCase(RRTestCase):
+    SUBNAME = '*.foobar'
+    TYPE = 'AAAA'
+    TTL = 12
+    CONTENT_VALUES = ['2001:fb24:45fd:d51:7937:b375:9cf3:5c62', '2001:ed06:5ebc:9d:87a:ce9f:1ceb:996',
+                      '2001:aa22:60e8:cec5:5650:9ff9:9a1b:b588', '2001:3ca:d710:52c2:9748:eec6:2e20:af0b',
+                      '2001:9c6e:8417:3c06:dd1c:44f1:a35f:ffad', '2001:f67a:5847:8dc0:edc3:56f3:a067:f80e',
+                      '2001:4e21:bda6:a509:e777:91c6:2dc1:394', '2001:9930:b062:c38f:99f6:ce12:bb04:f7c6',
+                      '2001:bb5e:921:b17f:7c9b:afb6:9933:cc79', '2001:a861:7139:e21e:11e4:8782:242b:e2a2',
+                      '2001:eaa:ff53:c819:93e:437c:ccc8:330c', '2001:6a88:fb92:5b43:984b:b729:393b:f173']
+    ALT_CONTENT_VALUES = ['2001:2d03:6247:3494:b92e:d4a:2827:e2d', '2001:4b37:19d6:b66e:1aa1:db0f:98b5:d065',
+                          '2001:dbf1:e401:ace2:bc99:eb22:6e12:ec81', '2001:fa92:3564:7c3f:9995:2068:58bf:2a45',
+                          '2001:4c2c:c671:9f0c:600e:4eb6:672e:48c7', '2001:5d09:a6f7:594b:afa4:318a:6eda:3ec6',
+                          '2001:f33a:407c:f4e6:f886:dce2:6d08:d8ae', '2001:43c8:378d:7d37:92eb:fb0c:26b1:4998',
+                          '2001:7293:88c5:5405:fd1:7334:bb55:be20', '2001:c4b7:ae76:a9a2:ffb5:ba30:6874:a416',
+                          '2001:175f:7880:ef82:b65a:a472:14c9:a495', '2001:8c35:1566:4f53:c26a:c54:2c9f:1463']
+
+
+class TXTRRTestCase(RRTestCase):
+    SUBNAME = '_acme_challenge'
+    TYPE = 'TXT'
+    TTL = 876
+    CONTENT_VALUES = ['The quick brown fox jumps over the lazy dog',
```

Comment:
These need to have each value surrounded by extra `"` characters inside the string